lib/livebook_web/live/session_live.ex

```diff
@@ -1188,6 +1189,12 @@ defmodule LivebookWeb.SessionLive do
     {:noreply, socket}
   end
 
+  def handle_info({:env_vars_changed, env_vars}, socket) do
+    Session.put_env_vars(socket.assigns.session.pid, env_vars)
```

Comment:
We move the subscription an and message handling to the session too, otherwise `put_env_vars` is called once per user :D